macros/models/_project/balancer/balancer_protocol_fee_macro.sql

```diff
@@ -0,0 +1,167 @@
+{% macro 
+    balancer_protocol_fee_macro(
+        blockchain, version
+    ) 
+%}
+
+WITH pool_labels AS (
+        SELECT * FROM (
+            SELECT
+                address,
+                name,
+                ROW_NUMBER() OVER (PARTITION BY address ORDER BY MAX(updated_at) DESC) AS num
+            FROM {{ ref('labels_balancer_v2_pools') }}
+            WHERE blockchain = '{{blockchain}}'
+            GROUP BY 1, 2) 
+        WHERE num = 1
+    ),
+
+    prices AS (
+        SELECT
+            date_trunc('day', minute) AS day,
+            contract_address AS token,
+            decimals,
+            AVG(price) AS price
+        FROM {{ source('prices', 'usd') }}
+        WHERE blockchain = '{{blockchain}}'
+        {% if is_incremental() %}
+        AND minute >= DATE_TRUNC('day', NOW() - interval '7' day)
+        {% endif %}           
```

Comment:
can we update all incremental logic to be like [this](https://github.com/duneanalytics/spellbook/blob/main/macros/models/_sector/dex/uniswap_compatible_trades.sql#L29-L32)?

that allows us to control incremental logic universally and manage in one place. all you need to do is ensure the correct column is passed in (with alias if needed)

Reply:
i'll only comment here to keep clean

Reply:
thanks for the review, Jeff! this macro is great, please lmk if I used it properly